src/lib/encoder.ts

```diff
@@ -201,62 +253,62 @@ export const valueContentEncodingMap = {
     }
 }
 
-
 export const encoder = {
+    encodeValueType: (type: string, value: string): string => {
+
+        if (!valueTypeEncodingMap[type]) {
 
-    encodeValueType: (type, value) => {
+            throw new Error('Could not encode valueType: "' + type + '".')
 
-        if (!valueTypeEncodingMap[type]) { throw new Error('Could not encode valueType: "' + type + '".') }
+        }
 
-        return (value)
-            ? valueTypeEncodingMap[type].encode(value)
-            : value
+        return value ? valueTypeEncodingMap[type].encode(value) : value
 
     },
 
-    decodeValueType: (type, value) => {
+    decodeValueType: (type: string, value: string) => {
+
+        if (!valueTypeEncodingMap[type]) {
 
-        if (!valueTypeEncodingMap[type]) { throw new Error('Could not decode valueType: "' + type + '".') }
+            throw new Error('Could not decode valueType: "' + type + '".')
+
+        }
 
         if (value === '0x') return null
 
-        return (value)
-            ? valueTypeEncodingMap[type].decode(value)
-            : value
+        return value ? valueTypeEncodingMap[type].decode(value) : value
 
     },
 
-
-    encodeValueContent: (type, value) => {
+    encodeValueContent: (type: string, value: string): string | false => {
```

Comment:
@robertdavid010 is this output type valid? The code returns either a `string` or `false`. I'd prefer not to mix types if possible.